includes/class-smaily-for-wp-widget.php

```diff
@@ -98,11 +111,12 @@ public function widget( $args, $instance ) {
 	 * @return array
 	 */
 	public function update( $new_instance, $old_instance ) {
-		$instance                = $old_instance;
-		$instance['title']       = esc_textarea( $new_instance['title'] );
-		$instance['show_name']   = isset( $new_instance['show_name'] ) ? (bool) $new_instance['show_name'] : false;
-		$instance['success_url'] = esc_url( $new_instance['success_url'] );
-		$instance['failure_url'] = esc_url( $new_instance['failure_url'] );
+		$instance                  = $old_instance;
+		$instance['title']         = esc_textarea( $new_instance['title'] );
+		$instance['show_name']     = isset( $new_instance['show_name'] ) ? (bool) $new_instance['show_name'] : false;
+		$instance['success_url']   = esc_url( $new_instance['success_url'] );
+		$instance['failure_url']   = esc_url( $new_instance['failure_url'] );
+		$instance['autoresponder'] = isset( $new_instance['autoresponder'] ) ? $new_instance['autoresponder'] : '';
```

Comment:
At least sanitize autoresponder value to an integer. Or only update the value if it is one of existing autoresponder IDs. Current logic allows user to manually change the form field value using browser's developer tools and set it to anything they would like to.

Reply:
I went ahead and added validation for it.
Validation falls back to the previous option if autoresponder ID doesn't exist.
9860214e37